main.js

```diff
@@ -40,29 +46,61 @@ reset.onclick = () => {
    window.location.reload();
 }
 
+// if clicking on a bomb, all bombs in the field will be explosed
+function bombExplosion(tableSize) {
+    for (let i = 0; i < tableSize; ++i) {
+        for (let j = 0; j < tableSize; ++j) {
+            if (board[i][j] == 'bomb') {
+                document.getElementById(i + " " + j).innerText = "💣";
+            }
+        }
+    }
+}
+
+// for each click, the board will be verified 
+// in order to determine if it was correct completed
+// after that, it is determined the status of the game
+function checkBoard(tableSize) {
+    correctCompleted = 0;
+    for (let i = 0; i < tableSize; ++i) {
+        for (let j = 0; j < tableSize; ++j) {
+            if (board[i][j] == 'mined' || (board[i][j] == "bomb" && document.getElementById(i + " " +j).innerText == "🚩")) {
+                ++correctCompleted;
+            }
+        }
+    }
+    statusGame(tableSize);
+}
+
 // adds a flag if clicking the left button of the mouse
+// if a bomb is mined, the field will explode
+// updated the status of the game
 function completeBoard(event, cell, line, column, tableSize) {
     if (event.button == 0) {
         if (board[line][column] == 'bomb') {
-            // TODO:
-            // if clicking on a bomb, the game is over;
+            bombExplosion(tableSize);
+            endGameStatus = true;
         } else {
             mineField(line, column, tableSize);
         }
     } else if (event.button == 2 && board[line][column] != 'mined') {
         if (cell.innerText == "") {
             cell.innerText = "🚩";
+            --bombs.innerText;
         } else {
             cell.innerText = "";
+            ++bombs.innerText;
         }
     }
+    checkBoard(tableSize);
 }
 
 // recieves the number of lines and columns and creates the table
 // adds an event listener on each cell, letting the player mine
 function createTable(tableSize) {
     level.style.display = "none";
     document.body.style.backgroundColor = "black";
+    bombs.innerText = tableSize * 2;
```

Comment:
At lines 16, 23, and 30 you initialize this variable with some numbers, why are you reinitializing it here?